sql/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -197,10 +204,16 @@ private GroupShardsIterator getShardIterators(WhereClause whereClause,
         if (whereClause.partitions().size() > 0) {
             routingIndices = whereClause.partitions().toArray(new String[whereClause.partitions().size()]);
         } else {
-            routingIndices = concreteIndices;
+            routingIndices = concreteOpenIndices;
         }
         ClusterState state = clusterService.state();
 
+        // If only 1 partition is touched, and is not in the concrete indices, this should fail.
+        if (whereClause.partitions().size() == 1 && !isAnOpenIndex(routingIndices[0])) {
+            throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                "The current operation is not allowed on partition %s, which is closed.", routingIndices[0]));
```

Comment:
I think we should keep all select queries consistent in that they never fail. I think it would be confusing for users that depending on the WHERE clause sometimes queries fail and sometimes they just don't match anything. (E.g. if you have a single partition and that is closed, `SELECT * FROM t` would return nothing, but `SELECT * FROM t where p = 1` would cause a failure.

But maybe wait for @dobe input here before changing it